### DISCUSSION TITLE
When to use layouts and when not?

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/nuxt#18957] [General]
From watching a Mastering Nuxt lessson about layouts - https://masteringnuxt.com/nuxt3/lessons/understanding-layouts - I gained impression, that using `layouts` directory is always a good idea. Because Michael says in the video it is a "convention" and  most developers would probably seek for header/footer stuff in there.

So based on that I started diggig up common code from my `app.vue` files (where I naturally place it during my baby steps with Nuxt) and putting them in `layouts/default.vue` everywhere.

But today I have checked something in the docs and noticed [layouts page](https://nuxt.com/docs/guide/directory-structure/layouts) says:

`If you only have a single layout in your application, we recommend using app.vue instead.`

So far my Nuxt apps are rather simple and I haven't yet implemented anything where I would actually use more than one layout.

So I would like to start a discussion about that, to better understand the concepts and beliefs of more experienced Nuxt users and members the development team themselves. **Should I avoid creating default layouts** just to have them in the code "because they may come handy one day"? Would you mind sharing your point of view on that?

**Comment by TheAlexLichter** [comment]
From my POV it is pretty straightforward and as the docs say.

If you don't have the need for a layout, use `app.vue`. You'll have less overhead and less components as layout-related stuff will be tree-shaken.

If you need layouts later, you can easily migrate by "copying" the `app.vue` content to the `default.vue` layout and adapt or even fully remove `app.vue`.
